source/adapters/hip/enqueue.cpp

```diff
@@ -1618,25 +1618,55 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueUSMMemcpy2D(
     hipPointerAttribute_t srcAttribs{};
     hipPointerAttribute_t dstAttribs{};
 
+    // Determine if pSrc and/or pDst are system allocated pageable host memory.
     bool srcIsSystemAlloc{false};
     bool dstIsSystemAlloc{false};
-
-    hipError_t hipRes{};
-    // hipErrorInvalidValue returned from hipPointerGetAttributes for a non-null
-    // pointer refers to an OS-allocation, hence pageable host memory. However,
-    // this means we cannot rely on the attributes result, hence we mark system
-    // pageable memory allocation manually as host memory. The HIP runtime can
-    // handle the registering/unregistering of the memory as long as the right
-    // copy-kind (direction) is provided to hipMemcpy2DAsync for this case.
-    hipRes = hipPointerGetAttributes(&srcAttribs, (const void *)pSrc);
-    if (hipRes == hipErrorInvalidValue && pSrc)
+    // Error code hipErrorInvalidValue returned from hipPointerGetAttributes
+    // for a non-null pointer refers to an OS-allocation, hence we can work
+    // with the assumption that this is a pointer to a pageable host memory.
+    // Since ROCm version 6.0.0, the enum hipMemoryType can also be marked as
+    // hipMemoryTypeUnregistered explicitly to relay that information better.
+    // This means we cannot rely on any attribute result, hence we just mark
+    // the pointer handle as system allocated pageable host memory.
+    // The HIP runtime can handle the registering/unregistering of the memory
+    // as long as the right copy-kind (direction) is provided to hipMemcpy2D*.
+    hipError_t hipRet = hipPointerGetAttributes(&srcAttribs, pSrc);
```

Comment:
Is there a reason we no longer need to cast `pSrc` to `const void *`? 

Also is it necessary to change `hipRes` to `hipRet`? I would avoid naming changes if possible.

I also think changing the order of checks from `(hipRes == blah && pSrc)` to `(pSrc && hipRes == blah)` is unnecessary.

Reply:
> Is there a reason we no longer need to cast pSrc to const void *?
`pSrc` is passed as `const void *` argument to the `urEnqueueUSMMemcpy2D` function, so I just avoided the unnecessary cast.

> Also is it necessary to change hipRes to hipRet? I would avoid naming changes if possible.
My bad with the renaming. I meant to leave it as it was, meaning `hipRes`. Will change to how it was before, i.e. `(hipRes == blah && pSrc)`.

> I also think changing the order of checks from (hipRes == blah && pSrc) to (pSrc && hipRes == blah) is unnecessary.
I agree the current renamings and change of order are unnecessary noise, I just didn't pay enough attention after reworking. Thanks!

Reply:
Thanks Georgi, I agree with removing the cast if unnecessary

Reply:
The unnecessary noise should be resolved in commit [83c5eee](https://github.com/oneapi-src/unified-runtime/pull/1455/commits/83c5eeed097f7930a5003c8ba024abcb8d9eee43).